pkg/hive/manager_test.go

```diff
@@ -158,3 +162,46 @@ func TestResetCorrelationData(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateOrUpdate(t *testing.T) {
+	subscriptionDoc := &api.SubscriptionDocument{
+		Subscription: &api.Subscription{
+			Properties: &api.SubscriptionProperties{
+				TenantID: "tenantID",
+			},
+		},
+	}
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+	ocDoc := &api.OpenShiftClusterDocument{
+		ID: "id",
+		OpenShiftCluster: &api.OpenShiftCluster{
+			ID:       resourceID,
+			Name:     "name",
+			Location: "location",
+			Properties: api.OpenShiftClusterProperties{
+				InfraID: "infraID",
+				HiveProfile: api.HiveProfile{
+					Namespace: "namespace",
+				},
+				NetworkProfile: api.NetworkProfile{
+					APIServerPrivateEndpointIP: "apiServerPrivateEndpointIP",
+				},
+			},
+		},
+	}
+
+	controller := gomock.NewController(t)
+	dh := mock_dynamichelper.NewMockInterface(controller)
+	cm := clusterManager{dh: dh}
+
+	dh.EXPECT().Ensure(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
```

Comment:
I think we either need to drop this test as it is not very useful or do some checking on what we receive here. Checking will be tricky.

Reply:
I am not sure what can be checked here. I guess you mean mocking `restcli` inside `dynamicHelper.ensureOne` (and checking for expected API calls), do you?

Reply:
If you were able to mock the call to `func (hr *clusterManager) resources(sub *api.SubscriptionDocument, doc *api.OpenShiftClusterDocument) ([]kruntime.Object, error) {` you'd be able to have something that you can actually check , (and thus replace the 4 `gomock.Any()` with some proper values). Not sure if it is feasible though

Reply:
I honestly believe covering the `clusterManager` with unit tests does not make much sense. It would be better suited for integration test, when it actually checks that it calls the stuff OK.